### DISCUSSION TITLE
Some XML syntax issues in localization files for item identifiers with non-standard naming conventions.

### DISCUSSION THREAD

**Discussion by lIllIIlI** [FakeFishGames/Barotrauma#14777] [Bug reports]
### Disclaimers

- [X] I have searched the issue tracker to check if the issue has already been reported.
- [X] My issue happened while using mods.

### What happened?

Lets say there is an item defined as follows:

`<Item name="普通7.92*57" identifier="pt7.92*57"  ...`

Above item appears correctly in native language (chinese), but when trying to localize said item into another language:

Xml breaks due to syntax issues.
![VSCodium_95KysEgpaL](https://github.com/user-attachments/assets/6301586d-91b3-4083-94d9-ba678dd7cd5d)

According to my xml parser, * and . are illegal and cannot be part of tag names/cannot be escaped? Correct me if I am wrong

![notepad++_vS7OHdlCZN](https://github.com/user-attachments/assets/b0a4d57b-aaf8-4430-923e-58bb0c6393e0)

Now, there are several dozen items named this way in this mod (Kyushu Arsenal), and currently I dont see any solution except having to override the identifiers and replace all * and . across dozens of files

### Reproduction steps

Create an item with identifier containing invalid characters (works fine for native language)
Try to localize said item in localization xml
XML throws syntax errors

### Bug prevalence

Just once

### Single player or multiplayer?

Happens outside single player or multiplayer game modes (e.g. game launches on startup, something broken in the main menu)

### Version

v1.5.9.2 (Summer Update Hotfix 2)

### Which operating system did you encounter this bug on?

Windows

**Comment by Regalis11** [comment]
Thank you for the report!

`*` is indeed not allowed in the names of the XML elements. `.` should be though, and it's also used in the vanilla game in some places. I'm afraid you probably need to change the identifiers.

I would not really consider this an issue with the game though: the game handles the XML correctly, the issue here is just that the XML is invalid.
